app/helpers/lookup_keys_helper.rb

```diff
@@ -24,7 +24,8 @@ def show_puppet_class(f)
       text_f(f, :puppetclass_id, :label => _('Puppet Class'), :value => f.object.param_class, :disabled => true)
     else # new smart-var with no particular context
          # Give a select for choosing the parent puppetclass
-      select_f(f, :puppetclass_id, Puppetclass.all, :id, :to_label, {}, {:label => _("Puppet class")})
+      puppetclasses = accessible_resource(f.object, :puppetclass)
+      select_f(f, :puppetclass_id, puppetclasses, :id, :to_label, { :include_blank => true }, {:label => _("Puppet class")})
```

Comment:
it was only in the logs because the validation was on puppetclass and not puppetclass_id and the field is on puppetclass_id